mpact_api/mpact/views.py

```diff
@@ -1,104 +1,47 @@
-from rest_framework import serializers
-from rest_framework.response import Response
+import asyncio
+
+from django.http import HttpResponse
 from rest_framework.views import APIView
 
-from .constants import MESSAGE, NEW_CHAT_PARTICIPANT, NEW_CHAT_TITLE, OK, TEXT
-from .logger import logger
-from .services import anonymize, set_webhook
-from .telegramevents import TelegramEvents
+from .services import get_dialogs, login, logout, send_msg
+
+loop = asyncio.get_event_loop()
```

Comment:
This will set `loop` when the module is imported.

You can create a new event loop, if one doesn't already exist, with this function, and call it from your view methods:
```suggestion
def new_or_current_event_loop():
    try:
        return asyncio.get_event_loop()
    except RuntimeError:
        loop = asyncio.new_event_loop()
        asyncio.set_event_loop(loop)
        return loop
```

Reply:
I tried this, it is always going in except block and it keeps on loading, it is not producing the result we want.

Reply:
The problem is happening because `services.py` is trying to use this event loop to start the "bot" client. You don't really want to start your clients outside of a function or a method either. I'll add some comments closer to the code ...